src/Raven.Client/Documents/Operations/AI/Agents/RunChatOperation.cs

```diff
@@ -11,47 +11,76 @@
 
-    public StartChatOperation(string identifier, string prompt, Dictionary<string, object> parameters = null)
+    private readonly string _chatId;
+    private readonly List<ToolResponse> _toolResponses;
+    public RunChatOperation(string identifier, string userPrompt, Dictionary<string, object> parameters)
     {
         ValidationMethods.AssertNotNullOrEmpty(identifier, nameof(identifier));
-        ValidationMethods.AssertNotNullOrEmpty(prompt, nameof(prompt));
+        ValidationMethods.AssertNotNullOrEmpty(userPrompt, nameof(userPrompt));
 
         _identifier = identifier;
-        _prompt = prompt;
+        _userPrompt = userPrompt;
         _parameters = parameters;
     }
 
+    public RunChatOperation(string chatId, string userPrompt = null, List<ToolResponse> toolResponses = null)
+    {
+        ValidationMethods.AssertNotNullOrEmpty(chatId, nameof(chatId));
+     
+        _chatId = chatId;
+        _userPrompt = userPrompt;
+        _toolResponses = toolResponses;
+    }
+
     public RavenCommand<ChatResult<TSchema>> GetCommand(DocumentConventions conventions, JsonOperationContext context)
     {
-        return new StartChatOperationCommand(_identifier, _prompt, _parameters, conventions);
+        return new RunChatOperationCommand(_chatId, _identifier, _userPrompt, _parameters, _toolResponses, conventions);
     }
 
-    internal sealed class StartChatOperationCommand : RavenCommand<ChatResult<TSchema>>
+    internal sealed class RunChatOperationCommand : RavenCommand<ChatResult<TSchema>>
     {
+        private readonly string _chatId;
         private readonly string _identifier;
         private readonly string _prompt;
         private readonly Dictionary<string, object> _parameters;
+        private readonly List<ToolResponse> _toolResponses;
         private readonly DocumentConventions _conventions;
 
-        public StartChatOperationCommand(string identifier, string prompt, Dictionary<string, object> parameters, DocumentConventions conventions)
+        public RunChatOperationCommand(string chatId, string identifier, string prompt, Dictionary<string, object> parameters, List<ToolResponse> toolResponses, DocumentConventions conventions)
         {
+            _chatId = chatId;
             _identifier = identifier;
             _prompt = prompt;
             _parameters = parameters;
+            _toolResponses = toolResponses;
             _conventions = conventions;
         }
         public override bool IsReadRequest => false;
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
-            url = $"{node.Url}/databases/{node.Database}/ai/agent/start?id={Uri.EscapeDataString(_identifier)}";
-            var body = new StartChatBody { Prompt = _prompt, Parameters = _parameters };
+            url = $"{node.Url}/databases/{node.Database}/ai/agent";
+            if (string.IsNullOrEmpty(_identifier) == false)
```

Comment:
How can this be null?
We skip it if there is a `chatId`?

Reply:
better to also call it `agentId` - although that too can be complex with whether this is `agentId` or `chatId` and if they are different.